iamy.go

```diff
@@ -40,16 +40,19 @@ const cloudformationStackNameTag = "aws:cloudformation:stack-name"
 
 func main() {
 	var (
-		debug         = kingpin.Flag("debug", "Show debugging output").Bool()
-		skipCfnTagged = kingpin.Flag("skip-cfn-tagged", fmt.Sprintf("Shorthand for --skip-tagged %s", cloudformationStackNameTag)).Bool()
-		skipTagged    = kingpin.Flag("skip-tagged", "Skips IAM entities (or buckets associated with bucket policies) tagged with a given tag").Strings()
-		includeTagged = kingpin.Flag("include-tagged", "Includes IAM entities (or buckets associated with bucket policies) tagged with a given tag").Strings()
-		pull          = kingpin.Command("pull", "Syncs IAM users, groups and policies from the active AWS account to files")
-		pullDir       = pull.Flag("dir", "The directory to dump yaml files to").Default(defaultDir).Short('d').String()
-		canDelete     = pull.Flag("delete", "Delete extraneous files from destination dir").Bool()
-		lookupCfn     = pull.Flag("accurate-cfn", "Fetch all known resource names from cloudformation to get exact filtering").Bool()
-		push          = kingpin.Command("push", "Syncs IAM users, groups and policies from files to the active AWS account")
-		pushDir       = push.Flag("dir", "The directory to load yaml files from").Default(defaultDir).Short('d').ExistingDir()
+		debug              = kingpin.Flag("debug", "Show debugging output").Bool()
+		skipCfnTagged      = kingpin.Flag("skip-cfn-tagged", fmt.Sprintf("Shorthand for --skip-tagged %s", cloudformationStackNameTag)).Bool()
+		skipTagged         = kingpin.Flag("skip-tagged", "Skips IAM entities (or buckets associated with bucket policies) tagged with a given tag").Strings()
+		includeTagged      = kingpin.Flag("include-tagged", "Includes IAM entities (or buckets associated with bucket policies) tagged with a given tag").Strings()
+		pull               = kingpin.Command("pull", "Syncs IAM users, groups and policies from the active AWS account to files")
+		pullDir            = pull.Flag("dir", "The directory to dump yaml files to").Default(defaultDir).Short('d').String()
+		pullCanDelete      = pull.Flag("delete", "Delete extraneous files from destination dir").Bool()
```

Comment:
I wonder whether we should offer the --delete flag for the normalise command, I can see someone wanting to do a "full" cleanup.

Reply:
We do: line 55

Reply:
Ahh, I should read more

Reply:
I'm torn as to whether to duplicate it like this, or to push it up to kingpin and tolerate the ability to uselessly supply it to the push command.

Reply:
I used `pushDir`/`pullDir` as the precedent, noting that the fact that `dryRun` is global meant I had to write a failure case to indicate that it's not supported. If I did anything more here, I'd:

1. Hoist `dir` to kingpin
2. Lower `dryRun` to `push`/`pull`, and
3. Leave `delete` where it is

I'm personally quite fussy about CLI behaviour, but I'll happily defer to the ideal choices for the application.